### PR TITLE
exposing additional functions for vault mock_crust tests

### PR DESCRIPTION
**Pull request by maqi**
This change is [Reviewable](https://reviewable.io/reviews/maidsafe/routing/1048)

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @afck

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: src/core_tests.rs [hunk 1]
```diff
@@ -337,29 +337,36 @@ fn entry_names_in_bucket(table: &RoutingTable, bucket_index: usize) -> HashSet<X
 
 // Get names of all nodes that belong to the `index`-th bucket in the `name`s
 // routing table.
-fn node_names_in_bucket(nodes: &[TestNode],
-                        name: &XorName,
+fn node_names_in_bucket(routing_tables: &Vec<RoutingTable>,
```
Comment (dirvine) [inline src/core_tests.rs]: why Vec here instead of `&[RoutingTable]` ?

**Comment by maqi** [comment]
*[src/core_tests.rs, line 340 \[r1\]](https://reviewable.io:443/reviews/maidsafe/routing/1048#-KK9Zp3ETPOQh6P0dvxd-r1-340:-KKCZ6kvnGNaZ7zHBHbE:1847131133) ([raw file](https://github.com/maidsafe/routing/blob/5c476a526fb8ee1e07da5d8fc60e634080a934f8/src/core_tests.rs#L340)):*
<details><summary><i>Previously, dirvine (David Irvine) wrote…</i></summary>
> why Vec here instead of `&[RoutingTable]` ?
</details>
resolved

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/maidsafe/routing/1048)*

**Comment by afck** [comment]
Review status: 0 of 4 files reviewed at latest revision, 3 unresolved discussions.

---

*[src/core_tests.rs, line 379 \[r2\]](https://reviewable.io:443/reviews/maidsafe/routing/1048#-KKCbpJUERY8H8MMplz1:-KKCbpJUERY8H8MMplz2:1529689191) ([raw file](https://github.com/maidsafe/routing/blob/926f5c54ff99b4bd7a4921c828fcb518a256af98/src/core_tests.rs#L379)):*
> ```Rust
> 
> // Verify that the kademlia invariant is upheld for all nodes.
> fn verify_kademlia_invariant_for_all_nodes(nodes: &[TestNode]) {
> ```

I'd expose that function, too, as in most cases we will probably want to check it for all nodes.

---

*[src/node.rs, line 129 \[r2\]](https://reviewable.io:443/reviews/maidsafe/routing/1048#-KKCcQX7JtpQbxxPx5RF:-KKCcQX8_m1jvUX4k2pg:-1592570724) ([raw file](https://github.com/maidsafe/routing/blob/926f5c54ff99b4bd7a4921c828fcb518a256af98/src/node.rs#L129)):*
> ```Rust
>     /// Routing table of this node.
>     pub fn routing_table(&self) -> RoutingTable {
>         self.core.borrow().routing_table().clone()
> ```

Instead of `clone`, `to_names` should be sufficient here:
http://docs.maidsafe.net/kademlia_routing_table/master/kademlia_routing_table/struct.RoutingTable.html#method.to_names
Then we don't need to expose the `NodeInfo` at all.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/maidsafe/routing/1048)*

**Comment by afck** [comment]
Reviewed 1 of 4 files at r1, 3 of 3 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/maidsafe/routing/1048)*

**Comment by maqi** [comment]
*[src/core_tests.rs, line 379 \[r2\]](https://reviewable.io:443/reviews/maidsafe/routing/1048#-KKCbpJUERY8H8MMplz1:-KKD22aKoMavDhjWhBzv:-1005538927) ([raw file](https://github.com/maidsafe/routing/blob/926f5c54ff99b4bd7a4921c828fcb518a256af98/src/core_tests.rs#L379)):*
<details><summary><i>Previously, afck (Andreas Fackler) wrote…</i></summary>
> I'd expose that function, too, as in most cases we will probably want to check it for all nodes.
</details>
but this is routing::TestNode, and vault has vault::TestNode
unless we make this and verify_kademlia_invariant_for_node both templated (or merge them into one templated function).

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/maidsafe/routing/1048)*

**Comment by Viv-Rajkumar** [comment]
closing this. @maqi can you raise this against master branch pls

**Pull request closed without merging**